core/validation/validate.py

```diff
@@ -9,11 +9,14 @@
 from numpy.typing import NDArray
 
 import core.validation.failures as vf
+import core.validation.historical as hvf
```

Comment:
Do we want to the file where the non-user facing failures are to be called historical.py, and to import it as hvf?

Reply:
I suppose it's not too important as long as it's clearly mentioned what we mean by historical in the file.

Reply:
I've done some renaming / refactoring of this in my latest commit 👍

Reply:
Yep, much clearer now, and a good way of organising our validation/failures